src/constraint/encoding.ts

```diff
@@ -140,6 +140,34 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
```

Comment:
This actually is affected by all scale's child properties.  
Don't type them manually, leverage `NESTED_ENCODING_PROPERTY_PARENT_INDEX`.